packages/restapi/src/lib/spaceV2/acceptInvite.ts

```diff
@@ -0,0 +1,301 @@
+                        });
+                    });
+                }
+
+                if (parsedData.type === 'endCall') {
+                    console.log('END CALL');
+
+                    if (
+                        parsedData?.details?.type === SPACE_DISCONNECT_TYPE.LEAVE
+                    ) {
+                        // destroy connection to only the current peer
+                        peerConnection?.destroy();
+                        peerConnection = null;
+                        this.setSpaceV2Data((oldData) => {
+                            return produce(oldData, (draft) => {
+                                const incomingIndex = getIncomingIndexFromAddress(
+                                    oldData.incomingPeerStreams,
+                                    recipientAddress
+                                );
+                                draft.incomingPeerStreams.splice(incomingIndex, 1);
+                            });
+                        });
+                    }
+                    if (
+                        parsedData?.details?.type === SPACE_DISCONNECT_TYPE.STOP
+                    ) {
+                        // destroy connection to all the peers
+                        for (const connectedAddress in this.getConnectedPeerIds()) {
+                            // TODO: refactor
+                            // this.getPeerConnection(connectedAddress)?.destroy();
+                            this.setPeerConnection(connectedAddress, undefined);
+                        }
+                    }
+
+                    if (
+                        parsedData?.details?.type === SPACE_DISCONNECT_TYPE.STOP
+                    ) {
+                        // destroy the local stream
+                        if (this.data.local.stream) {
+                            endStream(this.data.local.stream);
+                        }
+
+                        // reset the state
+                        this.setSpaceV2Data(() => initSpaceV2Data);
+                    }
+                }
+            } else {
+                onReceiveMessage(data);
+            }
+        });
+
+        peerConnection.on(
+            'stream',
+            (currentStream: MediaStream) => {
+                console.log('received incoming stream', currentStream);
+                const incomingIndex = getIncomingIndexFromAddress(
+                    this.data.pendingPeerStreams,
+                    recipientAddress
+                );
+
+                // Here, we can handle if we want to merge stream or anything
+                // this.onReceiveStream(
+                //   currentStream,
+                //   recipientAddress,
+                //   this.data.pendingPeerStreams[pendingIndex].audio
+                // );
+
+                this.setSpaceV2Data((oldData) => {
+                    return produce(oldData, (draft) => {
+                        draft.incomingPeerStreams[incomingIndex].stream = currentStream;
```

Comment:
peer is not removed from pendingPeerStreams
and at this point, peer is not added in incomingPeerStreams, so updating it would cause error